pkg/util/path-generator.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2023 LY Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"fmt"
+	"html/template"
+	"strings"
+)
+
+// GeneratePath generates the output path for the credentials by applying the domain name, role name, and delimiter to a specified naming format.
+// i.e) namingFormat=="/var/run/athenz/rolecerts/{{domain}}{{delimiter}}{{role}}"
+func GeneratePath(namingFormat, domain, role, delimiter string) (string, error) {
+	if namingFormat == "" {
+		return "", fmt.Errorf("naming format is empty")
+	}
+	if domain == "" {
+		return "", fmt.Errorf("domain is empty")
+	}
+	// If the role is an empty string, the delimiter used to separate the domain name and the role name is not necessary.
```

Comment:
```suggestion
	// If the role is an empty string, the delimiter used to separate the domain name and the role name is discarded as not necessary
```
or, since the word `delimiter` has the meaning `separate`, its kind of redundant.
```suggestion
	// If the role is an empty string, the delimiter is discarded as not necessary
```